src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php

```diff
@@ -367,7 +367,7 @@ protected function assignGroupBy($queryBuilder)
     protected function getJoins()
     {
         $joins = [];
-        $fields = \array_merge($this->sortFields, $this->selectFields, $this->searchFields, $this->expressionFields);
```

Comment:
I don't think we can remove the `$this->expressionFields` as when we have a `where` condition on a join which was not sorted or selected the where will fail: https://github.com/sulu/sulu/blob/a9a5deb1d306f36960a00e56f15e7d64c8819435/src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php#L553

Reply:
i havent had a deeper look at this yet to be honest. in theory, this `WHERE` condition should only be needed on the query that selects the ids (not on the the main query), right?

Reply:
i think the part of the code is only called in these two places, so its not used for the "main" query. but i will need to add an additional check for the first call
https://github.com/sulu/sulu/blob/7833cdced3878fe25303c0f93c99daeba1e5e6b8/src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php#L208-L213
https://github.com/sulu/sulu/blob/7833cdced3878fe25303c0f93c99daeba1e5e6b8/src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php#L426

Reply:
Yes. Seems like the case without limit also checks that there is no expression: https://github.com/sulu/sulu/blob/a9a5deb1d306f36960a00e56f15e7d64c8819435/src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php#L209-L212
So I think this is save. I would like to backport this change then to the `2.1` branch.

Reply:
at least it looks like there are no failing test cases. but i think i will need to add an additional parameter because we still need to join the search fields in this case 🙈 
https://github.com/sulu/sulu/blob/7833cdced3878fe25303c0f93c99daeba1e5e6b8/src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php#L208-L213

will backport the change onto 2.1 once it works here 🙂

Reply:
backported in #5943